### ISSUE TITLE
Error return when`SSHAccess` set to false

### ISSUE THREAD

**Issue by tedteng** [gardener/gardener#7481]
**How to categorize this issue?**

/area ops-productivity
/area security
/kind bug

**What happened**:
AWS cluster created in Dev landscape, then change shoot yaml to
```
    workersSettings:
      sshAccess:
        enabled: false
```

after reconcile 
```
Status:
  Last Error:
    Description:  Error reconciling infrastructure: failed to apply the terraform config: Terraform execution for command 'apply' could not be completed:

* error importing EC2 Key Pair (shoot--core--i333xxx-aws-ssh-publickey): MissingParameter: The request must contain the parameter PublicKeyMaterial
  status code: 400, request id: <omitted>
  with aws_key_pair.kubernetes,
  on main.tf line 357, in resource "aws_key_pair" "kubernetes":
 357: resource "aws_key_pair" "kubernetes" {
    Last Update Time:  2023-02-13T00:54:24Z
  Last Operation:
    Description:  Error reconciling infrastructure: failed to apply the terraform config: Terraform execution for command 'apply' could not be completed:
``` 
**What you expected to happen**:
sshAccess enabled 
**How to reproduce it (as minimally and precisely as possible)**:
```
    workersSettings:
      sshAccess:
        enabled: false
```
**Anything else we need to know?**:
K8S 1.25.5

**Environment**:

- Gardener version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by rfranzke** [comment]
cc @AleksandarSavchev @dimityrmirchev

**Comment by AleksandarSavchev** [comment]
Fixed by [MCM-AWS#104](https://github.com/gardener/machine-controller-manager-provider-aws/pull/104) and [AWS#704](https://github.com/gardener/gardener-extension-provider-aws/pull/704)
/close
